frontend/src/components/TypeaheadSelect.tsx

```diff
@@ -248,12 +246,17 @@ const TypeaheadSelect: React.FunctionComponent<TypeaheadSelectProps> = ({
   };
 
   const onTextInputChange = (_event: React.FormEvent<HTMLInputElement>, value: string) => {
-    setFilterValue(value || '');
+    setFilterValue(value);
     setIsFiltering(true);
     if (onInputChange) {
       onInputChange(value);
     }
-
+    if (selected && value !== selected.content) {
+      // Clear the selection when the input value changes
+      if (onSelect) {
+        onSelect(undefined, '');
+      }
+    }
```

Comment:
This seems possibly undesirable, the selection shouldn't change until the user clicks another option. I assume this was to fix the value being duplicated when there's a selected option that has a space and you type in the input? I wonder if there's another way to fix that.

Reply:
Actually @YuliaKrimerman I think we should remove these lines, I troubleshooted a little and I think maybe the problem is outside this component. If you console.log `selectOptions` in `RoleBindingPermissionsNameInput` you'll see that actually when that option gets duplicated it is really being added in the list of options we pass down here. So it's something strange we're doing out there. It's also very strange that it only seems to happen for items that have a space in them.

I think it has something to do with these lines in `RoleBindingPermissionsNameInput`:

```ts
  if (value && !typeAhead.includes(value)) {
    selectOptions.push({ value, content: value });
  }
```

Ah -- it's because of the `namespaceToProjectDisplayName` thing we do there. When you select a project that includes spaces, that's a display name like "savitha MR test", which is what ends up in our `selectOptions` and is selected as our `value` there. But the `typeAhead` array is actual project names, like "savitha-mr-test". So the above lines are triggering adding the option to the list because it thinks it's missing.

I think the fix is to change the above line to:

```ts
  if (value && !selectOptions.some((option) => option.value === value)) {
    selectOptions.push({ value, content: value });
  }
```

Reply:
![Screenshot 2024-09-19 at 2 03 59 PM](https://github.com/user-attachments/assets/7574ccc7-e8ea-421c-b2a1-75ce1d39c56f)
This is how it's behaving now when I implement it with the suggested fix. It does that and then after a few seconds it wipes out those funny duplicates and behave normally

Reply:
Addressed all other comments and kept the original implementation for this, let me know if it's ok to keep it, or if we want to take out this issue for the future.

Reply:
@YuliaKrimerman with your other change in place to remove the `selectOptions.push` code, I tried removing the lines you added here (clear the selection when the input value changes) and it seems to behave like I would expect. I'm not seeing the above duplicates. Can you try removing this now and see if it works for you?